business/istio_validations.go

```diff
@@ -96,35 +86,29 @@ func (in *IstioValidationsService) GetValidations(namespace, service string) (mo
 
 	objectCheckers := in.getAllObjectCheckers(namespace, istioConfigList, exportedResources, services, workloadsPerNamespace, workloadsPerNamespace[namespace], gatewaysPerNamespace, mtlsDetails, rbacDetails, namespaces, registryStatus)
 
-	if service != "" {
-		objectCheckers = append(objectCheckers, in.getServiceCheckers(namespace, services, deployments, pods)...)
-	}
-
 	// Get group validations for same kind istio objects
 	validations := runObjectCheckers(objectCheckers)
+
 	if service != "" {
+		// in.businessLayer.Svc.GetServiceList(criteria) on fetchServices performs the validations on the service
+		// No need to re-fetch deployments+pods for this
+		validations.MergeValidations(services.Validations)
```

Comment:
+1 on your comment @xeviknal, these changes were more to adjust the code with the refactoring, but your suggestions make sense and probably it can be included as part of the follow-up tasks from this work when the logic of the validations can be updated with the information fetched from the Istio Registry.

I also have same feeling, that the Service validations are less-contextual, and they only need the Service info, no need to check other Istio resource for it.